src/main/kotlin/com/ampnet/reportservice/service/impl/TemplateDataServiceImpl.kt

```diff
@@ -158,4 +176,26 @@ class TemplateDataServiceImpl(
                 ErrorCode.INT_REQUEST, "Transaction doesn't belong to user wallet with hash: $txHash"
             )
     }
+
+    private fun getTranslations(userLanguage: String): Translations {
+        val translations = mutableMapOf<String, String>()
+        allTranslations.forEach { (key, map) ->
```

Comment:
If I get it correct, each time you need Translations class there a loop through all translations keys to generate Translations class? This is not an optimal solution, we should be able to load and store Translations class for each language and return it without generating it each time.